packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -433,133 +659,238 @@ fn array_to_user_warnings(array: mv8::Value<'_>) -> Vec<UserWarning> {
     }]
 }
 
-fn get_js_error(_mv8: &MiniV8, return_val: &mv8::Object<'_>) -> Option<Error> {
-    if let Ok(errors) = return_val.get("user_errors") {
-        if !matches!(errors, mv8::Value::Undefined) && !matches!(errors, mv8::Value::Null) {
-            let errors = array_to_user_errors(errors);
+fn get_js_error<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    return_val: v8::Local<'s, v8::Object>,
+) -> Option<Error> {
+    tracing::debug!("Collecting JS errors");
+    let user_errors = if let Some(user_errors) = v8::String::new(scope, "user_errors") {
+        user_errors
+    } else {
+        return Some(Error::V8("Could not create String".to_string()));
+    };
+
+    if let Some(errors) = return_val.get(scope, user_errors.into()) {
+        if !errors.is_null_or_undefined() {
+            let errors = array_to_user_errors(scope, errors);
             if !errors.is_empty() {
                 return Some(Error::User(errors));
             }
         }
     }
 
-    if let Ok(mv8::Value::String(e)) = return_val.get("pkg_error") {
-        // TODO: Don't silently ignore non-string, non-null-or-undefined errors
-        //       (try to convert error value to JSON string and return as error?).
-        return Some(Error::Package(PackageError(e.to_string())));
+    let pkg_error = if let Some(pkg_error) = v8::String::new(scope, "pkg_error") {
+        pkg_error
+    } else {
+        return Some(Error::V8("Could not create String".to_string()));
+    };
```

Comment:
```suggestion
    let pkg_error = match new_js_string(scope, "pkg_error") {
        Ok(pkg_error) => pkg_error,
        Err(err) => return Some(err),
    };
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/201d0b37fcd1abad7223e7ca21de13d82660d8c7